pkg/java/pom/parse.go

```diff
@@ -172,22 +189,69 @@ func (p *parser) parseRoot(root artifact) ([]types.Library, []types.Dependency,
 			uniqArtifacts[art.Name()] = artifact{
 				Version:  art.Version,
 				Licenses: art.Licenses,
+				Direct:   art.Direct,
 			}
+
+			// save only dependency names
+			// version will be determined later
+			dependsOn := lo.FilterMap(result.dependencies, func(a artifact, _ int) (string, bool) {
+				return a.Name(), !slices.Contains(savedDeps, a.Name())
+			})
+			uniqDeps[art.Name()] = dependsOn
+			// mvn only takes top dependencies
+			// this is needed to reproduce mvn logic
+			// take a look at `soft requirement`, `soft requirement with transitive dependencies`
+			// and `hard requirement for the specified version` tests
```

Comment:
It makes sense for SBOM.
But i thought about versions for these dependencies ( i mean cases when version was overwritten).

I checked [maven Cyclonedx plugin](https://github.com/CycloneDX/cyclonedx-maven-plugin):
pom.xml dependencies:
```
    <dependencies>
       <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
            <version>2.15.0</version>
        </dependency>
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-core</artifactId>
            <version>2.15.2</version>
        </dependency>
    </dependencies>
```
maven dep tree:
```
[INFO] com.example:example:jar:1.0.0
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.15.0:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.15.0:compile
[INFO] \- com.fasterxml.jackson.core:jackson-core:jar:2.15.2:compile
```
dependencies from built (https://github.com/CycloneDX/cyclonedx-maven-plugin) SBOM file:
```
  "dependencies" : [
    {
      "ref" : "pkg:maven/com.example/example@1.0.0?type=jar",
      "dependsOn" : [
        "pkg:maven/com.fasterxml.jackson.core/jackson-databind@2.15.0?type=jar",
        "pkg:maven/com.fasterxml.jackson.core/jackson-core@2.15.2?type=jar"
      ]
    },
    {
      "ref" : "pkg:maven/com.fasterxml.jackson.core/jackson-databind@2.15.0?type=jar",
      "dependsOn" : [
        "pkg:maven/com.fasterxml.jackson.core/jackson-annotations@2.15.0?type=jar",
        "pkg:maven/com.fasterxml.jackson.core/jackson-core@2.15.2?type=jar"
      ]
    },
```

So we can use similar logic. 
I will update PR.

Reply:
Yeah, the cyclonedx plugin behaves as I suggested. `com.fasterxml.jackson.core:jackson-annotations:2.15.0` => `com.fasterxml.jackson.core:jackson-core:2.15.2` is hidden in `mvn dependency:tree`, but the SBOM report contains the relationship. As for your concern regarding version overriding, I guess `jackson-core@2.15.0` is overridden by `jackson-core@2.15.2`, and the SBOM describes `jackson-databind@2.15.0` depends on `jackson-core@2.15.2` instead of `jackson-core@2.15.0`. It makes sense to me.

Reply:
Changed in https://github.com/aquasecurity/go-dep-parser/pull/243/commits/a2af0ef8efec7344712adbbb30c8574b38875918